File: app/src/main/java/com/glucocontrol/domain/model/ReadingTag.kt
package com.glucocontrol.domain.model

/**
 * Categorías clínicas que contextualizan el momento de la medición.
 * Se usan para dar relevancia médica al valor numérico de glucosa.
 *
 * [label] es el texto visible en la UI.
 */
enum class ReadingTag(val label: String) {
    AYUNAS("Ayunas"),
    PRE_COMIDA("Pre-comida"),
    POST_COMIDA("Post-comida"),
    ANTES_DORMIR("Antes de dormir"),
    OTRO("Otro"),
}

File: lib/domain/model/reading_tag.dart
enum ReadingTag {
  ayunas('Ayunas'),
  preComida('Pre-comida'),
  postComida('Post-comida'),
  antesDormir('Antes de dormir'),
  otro('Otro');

  const ReadingTag(this.label);
  final String label;

  static ReadingTag fromString(String value) {
    return ReadingTag.values.firstWhere(
      (t) => t.name.toUpperCase() == value.toUpperCase(),
      orElse: () => ReadingTag.otro,
    );
  }
}
